### DISCUSSION TITLE
Rotate VPN IP Address every 24 hours

### DISCUSSION THREAD

**Discussion by yTallBalls** [qdm12/gluetun#1406] [Ideas]
I was wanting to do something along the lines of switching my VPN IP address every 24 hours.

For example, I use Mullvad for my VPN (connecting via wireguard) and have the multiple cities defined via SERVER_CITIES variable. Would like to rotate to different locations every 24 hours thus changing my VPN address. Is this possible or could it be added as a feature?

**Comment by dedworks** [comment]
+1

I am using a bash script and a cron job to force a new server and IP.

No matter how many values I provide for SERVER_CITIES and SERVER_HOSTNAMES, it always chooses the same (first?) one and Mullvad assigns me the same IP unless I disconnect for a long while. 

My script changes the `SERVER_HOSTNAMES` value to a single random server matching one of the `SERVER_CITIES`, and restarts the container.

Would love to hear another way to reliably get a new IP with gluetun and Mullvad. (Or get suggestions to improve the script.)

I'm using openvpn, not wireguard, but the script should work for both.

https://gist.github.com/dedworks/2c3def2ffde3dafd8494860b78b8b70b

**Reply by trollollol123** [reply]
I can't make this work. Does it still work for you?

No candidate servers returned for cities 'Stockholm,Gothenburg,Helsinki,Copenhagen

**Comment by paluigi** [comment]
+1
That would be a really nice option if could be available directly in gluetun!
